### PR TITLE
Run Damlc tests using standalone check tool

### PR DESCRIPTION
**Pull request by dylant-da**
This moves those tests in DamlcUpgrades and UpgradesSpecBase which can be run in the standalone check tool to use that instead.

### CODE REVIEW

File: sdk/compiler/damlc/tests/src/DA/Test/DamlcUpgrades.hs [hunk 1]
```diff
@@ -874,6 +619,33 @@ tests damlc =
             createDirectoryIfMissing True (takeDirectory $ dir </> file)
             writeFileUTF8 (dir </> file) content
 
+data TestOptions = TestOptions
```
Comment (paulbrauner-da) [inline sdk/compiler/damlc/tests/src/DA/Test/DamlcUpgrades.hs:622]: This makes things much more readable! Thanks.

File: sdk/daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/UpgradeCheckMain.scala [hunk 2]
```diff
@@ -57,13 +57,21 @@ object UpgradeCheckMain {
...
+        case Right(()) => 0
       }
     }
   }
+
+  def main(args: Array[String]) = sys.exit(check(args))
+}
+
+object UpgradeCheckMain {
+  def default: UpgradeCheckMain = {
```
Comment (paulbrauner-da) [inline sdk/daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/UpgradeCheckMain.scala:74]: Nit: in the canton codebase they tend to call that "instance".

Also, could it be a val, or a lazy val? It seems to be quite stateless so I don't think you don't need a new instance each time.

File: sdk/daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/upgrade/UpgradesSpecBase.scala [hunk 3]
```diff
@@ -69,26 +68,7 @@ class UpgradesSpecLedgerAPI(override val suffix: String = "Ledger API")
 }
 
 trait ShortTests { this: UpgradesSpec =>
```
Comment (paulbrauner-da) [inline sdk/daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/upgrade/UpgradesSpecBase.scala:70]: Should this disappear entirely?

File: sdk/daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/upgrade/UpgradesSpecBase.scala [hunk 4]
```diff
@@ -1049,6 +324,57 @@ abstract class UpgradesSpec(val suffix: String)
     testPackagePair(v1Upload, v2Upload, uploadAssertion)
   }
 
+  def testPackagePairUpgradeCheck(
```
Comment (paulbrauner-da) [inline sdk/daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/upgrade/UpgradesSpecBase.scala:327]: I think you can delete that function and the one below now.

**Review by paulbrauner-da — CHANGES_REQUESTED** [review]
Looking good! Not approving yet because I'd like your thoughts on why we can't call the main of the upgrade check tool directly from haskell too, instead of spawning a process.

File: sdk/compiler/damlc/tests/src/DA/Test/DamlcUpgrades.hs [hunk 5]
```diff
@@ -593,115 +398,55 @@ tests damlc =
...
-        -> a
-        -> Dependency
-        -> Bool
-        -> Bool
-        -> Bool
-        -> [String] -> [String]
-        -> TestTree
-    testGeneral name location expectation lfVersion sharedDep warnBadInterfaceInstances setUpgradeField doTypecheck additionalDarsV1 additionalDarsV2 =
-        let upgradeFieldTrailer = if not setUpgradeField then " (no upgrades field)" else ""
+    testUpgradeCheck name expectation =
```
[the remark this answers is not in the document]
Comment (dylant-da) [reply sdk/compiler/damlc/tests/src/DA/Test/DamlcUpgrades.hs:408]: The main issue right now is that these sorts of warnings and errors come from a few different places, and there isn't necessarily a very easy way of capturing all of them while keeping their structure - the only realistic place I can capture all the warnings is once they've all been turned into `Diagnostic`, at which point the errors have no structure and we'll need to use regices anyways.

File: sdk/daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/UpgradeCheckMain.scala (hunk 2 above)
Comment (dylant-da) [reply sdk/daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/UpgradeCheckMain.scala:74]: done

File: sdk/compiler/damlc/tests/src/DA/Test/DamlcUpgrades.hs (hunk 1 above)
Comment (dylant-da) [reply sdk/compiler/damlc/tests/src/DA/Test/DamlcUpgrades.hs:622]: :)

File: sdk/daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/upgrade/UpgradesSpecBase.scala (hunk 3 above)
Comment (dylant-da) [reply sdk/daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/upgrade/UpgradesSpecBase.scala:70]: Good catch, yes, done

File: sdk/daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/upgrade/UpgradesSpecBase.scala (hunk 4 above)
Comment (dylant-da) [reply sdk/daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/upgrade/UpgradesSpecBase.scala:327]: done

File: sdk/compiler/damlc/tests/src/DA/Test/DamlcUpgrades.hs (hunk 5 above)
[the remark this answers is not in the document]
Comment (paulbrauner-da) [reply sdk/compiler/damlc/tests/src/DA/Test/DamlcUpgrades.hs:408]: Fair, thanks for explaining.

**Review by paulbrauner-da — APPROVED** [review]

File: sdk/compiler/damlc/tests/src/DA/Test/DamlcUpgrades.hs (hunk 5 above)
[the remark this answers is not in the document]
Comment (dylant-da) [reply sdk/compiler/damlc/tests/src/DA/Test/DamlcUpgrades.hs:408]: To which the question might well be "the why not at least call the upgrade check tool lib directly, so we can do regexes directly over the diagnostics" to which I'd say that we can do that, sure, but the structure we get back is relatively little, and it'd require a mild refactor of the tests.

I think we should do a separate PR where I write some proper integration tests over the upgrade check tool, and we can do the refactor there where it's most appropriate.

**Pull request merged**